Show concurrency in the guided tour

amartini51 opened the pull request:
Fixes: rdar://78794648

etcwilde commented on TSPL.docc/GuidedTour/GuidedTour.md:
```diff
@@ -2257,6 +2257,158 @@ print(fridgeIsOpen)
+
+You structure concurrent code using task groups and child tasks.
+Writing `async`-`let` implicitly creates a child task.
+
```
This example seems like it could you a touch more description. Maybe something about "creates a child task for each recipe and collects the baked desert at the end"?

amartini51 replied:
The pattern in the guided tour is to show the reader a sample of (most of) the language features in Swift and their syntax, without going into really any level of explanation.  None of the other code listings have text around them that explains what the code is doing.  So I'd rather leave this as is.

etcwilde commented on TSPL.docc/GuidedTour/GuidedTour.md:
```diff
@@ -2257,6 +2257,158 @@ print(fridgeIsOpen)
+
+Actors are similar to classes,
+except they ensure that different asynchronous functions
+can all interact with an instance of the same actor at the same time.
```
"... at the same time, safely."

Technically people could call methods on classes from different dispatch queues and threads before, but then they'd run into  data races.

amartini51 replied:
Yes, this is worth pointing out.

Changed in b2bbaaca.

amartini51 commented:
@etcwilde Can you please take another look?

After I converted the PR from a draft to "ready for review", I realized that the tour already had a bunch of this content, so with the PR's changes concurrency ended up being described twice.  I put this back in "draft" state, but there wasn't a way to un-request review.

I removed the duplicates, and changed the code listings so they follow the same running example used for the existing concurrency discussion in the guide.

etcwilde reviewed (approved):
Yep, this looks reasonable to me.
